src/vos/lru_array.h

```diff
@@ -96,18 +98,20 @@ lrua_remove_entry(struct lru_entry *entries, struct lru_entry *entry)
 
 /** Internal API: Insert an entry in the lru list */
 static inline void
-lrua_insert_entry(struct lru_entry *entries, struct lru_entry *entry,
-		  uint32_t idx, uint32_t prev_idx, uint32_t next_idx)
+lrua_insert_mru(struct lru_array *array, struct lru_entry *entry,
+		uint32_t idx)
 {
+	struct lru_entry	*entries = &array->la_table[0];
 	struct lru_entry	*prev;
 	struct lru_entry	*next;
 
-	prev = &entries[prev_idx];
-	next = &entries[next_idx];
+	prev = &entries[array->la_mru];
+	next = &entries[array->la_lru];
+	D_ASSERT(prev != next);
```

Comment:
What will happen if there is single entry in the cache? still "prev != next"?

Reply:
This can only happen if the user allocates an lru array with 2 or fewer entries...I can add a check in initialization for that.